frappe/core/doctype/user/user.py

```diff
@@ -233,7 +233,7 @@ def has_website_permission(self, ptype, user, verbose=False):
 		return self.name == frappe.session.user
 
 	def set_full_name(self):
-		self.full_name = " ".join(filter(None, [self.first_name, self.last_name]))
+		self.full_name = " ".join(filter(None, [self.first_name, self.middle_name, self.last_name]))
```

Comment:
This is not the norm in most countries. Also, we are using this in UI everywhere, we don't need middle name to be displayed here. 

If you need it, you can add a simple server script. 


---

Side note: There are also a significant minority of countries where last name comes first in "full name". 

Alternate design: Let users modify full name to whatever they want. This full name should only be set if user hasn't set anything.